src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -656,12 +656,20 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref HashS
                     case TypeKind.Struct:
                     case TypeKind.Interface:
                     case TypeKind.Delegate:
-                        foreach (var typeArg in ((NamedTypeSymbol)current).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics)
+                        var namedType = (NamedTypeSymbol)current;
+                        if (namedType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics.IsEmpty)
+                        {
+                            return null;
+                        }
+
+                        int i;
+                        for (i = 0; i < namedType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics.Length - 1; i++)
```

Comment:
>namedType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics [](start = 40, length = 58)

Consider cashing this in a local. Who knows how expensive it could be for some implementations.

Reply:
It looks like common implementation allocates a new array on every access.

---
In reply to: [443271903](https://github.com/dotnet/roslyn/pull/45328#discussion_r443271903) [](ancestors = 443271903)